pkg/render/apiserver.go

```diff
@@ -1309,13 +1311,11 @@ func (c *apiServerComponent) tigeraApiServerClusterRoleBinding() *rbacv1.Cluster
 }
 
 // tierGetterClusterRole creates a clusterrole that gives permissions to get tiers.
-//
-// Calico Enterprise only
 func (c *apiServerComponent) tierGetterClusterRole() *rbacv1.ClusterRole {
 	return &rbacv1.ClusterRole{
 		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "tigera-tier-getter",
```

Comment:
For enterprise clusters on upgrade, this is going to result in two duplicated ClusterRole objects.

We need some explicit logic to delete the old `tiger-tier-getter` named object to ensure we don't leave unused resources around.

Reply:
Is it better to leave the name unchanged?